### ISSUE TITLE
Supporting for detection of skipping heading levels

### ISSUE THREAD

**Issue by kagankan** [markuplint/markuplint#695]
I would like to detect if the heading levels are used in the correct order.

## Incorrect examples

```html
<!-- incorrect: skipping h2 -->
<h1>Heading level 1</h1>
<h3>Heading level 3</h3>
<h4>Heading level 4</h4>
```

```html
<!-- incorrect: starting with h2 -->
<h2>Heading level 2</h2>
<h1>Heading level 1</h1>
<h2>Heading level 2</h2>
```

## Correct example

```html
<!-- correct: starting with h1 and heading levels increase by one -->
<h1>Heading level 1</h1>
<h2>Heading level 2</h2>
<h3>Heading level 3</h3>
<h4>Heading level 4</h4>
<h2>Heading level 2</h2>
```

## Ideas of rule name

- `heading-order`
- `no-skipping-heading-levels`

## References

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements#accessibility_concerns
https://dequeuniversity.com/rules/axe/4.6/heading-order
